cassandra/src/main/java/org/apache/calcite/adapter/cassandra/CassandraFilter.java

```diff
@@ -174,14 +179,28 @@ private String translateMatch(RexNode condition) {
       }
     }
 
-    /** Convert the value of a literal to a string.
+    /** Returns the value of the literal.
      *
      * @param literal Literal to translate
-     * @return String representation of the literal
+     * @return The value of the literal in the form of the actual type.
      */
-    private static String literalValue(RexLiteral literal) {
-      Object value = literal.getValue2();
-      return String.valueOf(value);
+    private static Object literalValue(RexLiteral literal) {
+      Comparable value = RexLiteral.value(literal);
+      String dateTimeFormatISO = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+      switch (literal.getTypeName()) {
+      case TIMESTAMP:
+      case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+        assert value instanceof TimestampString;
+        final SimpleDateFormat dateFormatter =
+            new SimpleDateFormat(dateTimeFormatISO, Locale.ROOT);
+        dateFormatter.setTimeZone(DateTimeUtils.UTC_ZONE);
```

Comment:
Are there any codes we can reuse in `DataTimeUtils` ?

Reply:
> Are there any codes we can reuse in `DataTimeUtils` ?

I did not find totally suitable code from DateTimeUtils. But I find that creating a SimpleDateFormat with setting time zone is used in several places in Calcite. So I add utility methods for creating SimpleDateFormat to the DateTimeStringUtils.